packages/app/src/cli/services/function/ui/components/Replay.tsx

```diff
@@ -0,0 +1,168 @@
+          marginY={1}
+          paddingTop={1}
+          flexDirection="column"
+          flexGrow={1}
+          borderStyle="single"
+          borderBottom={false}
+          borderLeft={false}
+          borderRight={false}
+          borderTop
+        >
+          {canUseShortcuts ? (
+            <Box flexDirection="column">
+              <Box flexDirection="row">
+                <Text>
+                  {figures.pointerSmall} {statusMessage}
+                </Text>
+              </Box>
+              <StatsDisplay recentFunctionRuns={recentFunctionRuns} />
+              <Text>
+                {figures.pointerSmall} Press <Text bold>q</Text> {figures.lineVertical} quit
+              </Text>
+            </Box>
+          ) : null}
+          {error ? <Text color="red">{error}</Text> : null}
+        </Box>
+      ) : null}
+    </>
+  )
+}
+
+function InputDisplay({input}: {input: string}) {
+  return (
+    <Box flexDirection="column">
+      <Text color="black" backgroundColor="yellow">
+        {'\n\n            Input            \n'}
+      </Text>
+      <Text>{prettyPrintJsonIfPossible(input)}</Text>
+    </Box>
+  )
+}
+
+function LogDisplay({logs}: {logs: string}) {
+  return (
+    <Box flexDirection="column">
+      <Text color="black" backgroundColor="blueBright">
+        {'\n\n            Logs            \n'}
+      </Text>
+      <Text>{logs}</Text>
+    </Box>
+  )
+}
+
+function OutputDisplay({output}: {output: string}) {
+  return (
+    <Box flexDirection="column">
+      <Text color="black" backgroundColor="greenBright">
+        {'\n\n           Output           \n'}
+      </Text>
+      <Text>{prettyPrintJsonIfPossible(output)}</Text>
+    </Box>
+  )
+}
+
+function BenchmarkDisplay({functionRun}: {functionRun: FunctionRun}) {
+  return (
+    <Box flexDirection="column">
+      <Text color="black" backgroundColor="green">
+        {'\n\n     Benchmark Results      \n'}
+      </Text>
+      <Text>Name: {functionRun.name}</Text>
+      <Text>Linear Memory Usage: {functionRun.memory_usage}KB</Text>
+      <Text>Instructions: {functionRun.instructions / 1000}K</Text>
+      <Text>Size: {functionRun.size}KB</Text>
+    </Box>
+  )
+}
+
+function StatsDisplay({recentFunctionRuns}: {recentFunctionRuns: [FunctionRun, FunctionRun]}) {
+  const delta = recentFunctionRuns[0].instructions - recentFunctionRuns[1].instructions
+  return (
+    <Box flexDirection="column">
+      <Text>
+        {figures.pointerSmall} Instruction count delta: {delta}
+      </Text>
+    </Box>
+  )
+}
+
+function ReplayLog({log}: {log: ReplayLog}) {
+  if (log.type === 'functionRun') {
+    return (
+      <Box flexDirection="column">
+        <InputDisplay input={log.input} />
+        <LogDisplay logs={log.logs} />
+        <OutputDisplay output={log.output} />
+        <BenchmarkDisplay functionRun={log} />
+      </Box>
+    )
+  }
+
+  if (log.type === 'systemMessage') {
+    return <Text>{log.message}</Text>
+  }
+
+  return null
```

Comment:
Nit: This could be a switch-case

```suggestion
  switch(log.type) {
    case "functionRun":
      return (
        <Box flexDirection="column">
          <InputDisplay input={log.input} />
          <LogDisplay logs={log.logs} />
          <OutputDisplay output={log.output} />
          <BenchmarkDisplay functionRun={log} />
        </Box>
      )
    case "systemMessage":
      return <Text>{log.message}</Text>
    default:
      return null;
  }
```